src/assets/Generator.Shared/HttpPipelineExtensions.cs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    internal static class HttpPipelineExtensions
```

Comment:
@AlexanderSher, what is the thinking behind putting `ProcessMessage` in an extension method, rather than on the Pipeline directly?  Will this logic eventually migrate into Pipeline?